tests/layering/helpers_test.go

```diff
@@ -0,0 +1,611 @@
+package e2e_test
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"math/rand"
+	"os"
+	"path"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	buildv1 "github.com/openshift/api/build/v1"
+	imagev1 "github.com/openshift/api/image/v1"
+	machineClient "github.com/openshift/client-go/machine/clientset/versioned"
+	"github.com/openshift/machine-config-operator/test/framework"
+	"github.com/openshift/machine-config-operator/test/helpers"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	aggregatedErr "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/kubernetes/pkg/credentialprovider"
+)
+
+// Gets the RHCOS base image tag from the BRANCH environment variable or
+// defaults to "latest"; e.g., registry.ci.openshift.org/rhcos-devel/rhel-coreos:<tag>
+func getBaseImageTag() string {
+	branch, found := os.LookupEnv("BRANCH")
+	if !found || branch == "" {
+		return "latest"
+	}
+
+	return strings.ReplaceAll(branch, "release-", "")
+}
+
+// Builds a derived OS image and places it into an Imagesteam. Attempts to use
+// a previously-built image, if one is available.
+func maybeBuildDerivedOSImage(t *testing.T, cs *framework.ClientSet) error {
+	imageStreamConfig := &imagev1.ImageStream{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      imageStreamName,
+			Namespace: mcoNamespace,
+		},
+	}
+
+	_, err := cs.ImageStreams(mcoNamespace).Create(context.TODO(), imageStreamConfig, metav1.CreateOptions{})
```

Comment:
Might be my imagination but it feels like a slightly more common pattern to call `Get` first and then `Create` second if it doesn't exist

Reply:
I could see that being the case. If that's what's more common, I'll adopt that pattern instead.

Reply:
I dunno if it actually is though, I bet @kikisdeliveryservice would know?

Reply:
I'm a little confused bc I can't find this line anymore - but it depends on what you're doing (which I can't tell bc I can't find the line)